src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -139,17 +139,22 @@ void AdaptDispatch::_WriteToBuffer(const std::wstring_view string)
 
         if (isWrapping)
         {
-            if (wrapAtEOL)
+            // We want to wrap, but we're not allowed to and we failed to write
+            // even a single character into the row. This can mean two things:
+            // * The glyph is wider than the buffer and can never be inserted.
```

Comment:
```suggestion
            // * The glyph is wider than the buffer and can never be inserted (which is indicated by the start column being 0)
```

Hey, what happens if you set a DECLRMM (@j4james PR #14876) that is 1 column wide and try to put a wide char in it?

Reply:
The left margin has to be less than the right margin, which guarantees a margin area at least two columns wide. Unless the screen itself is only wide column wide, but then it should behave as if no margins are set at all, which would assumedly be covered by the fix in this PR.

Reply:
I think I would feel a little more comfortable if it was hardened against this anyways. I think this can be achieved by replacing the condition with `state.columnBegin == leftMargin`.

Reply:
I don't think that's technically correct, though, because the left margin isn't necessarily where the cursor will end up when it wraps. That won't matter if the only time this can occur is when the left margin is 0 anyway, but then it's just making the code more confusing. Because either it's not needed, or it's wrong.

Reply:
Hmm I haven't read your PR yet, so I'm not sure I understand that. The reason I'm worried about this is because I'm playing around with the idea of introducing cells that are wider than 2 columns for the purpose of improved Unicode correctness during reflow on resize. I'm not sure yet whether it'll work out since most (all?) other terminals don't do that, but I would like to avoid issues like this one, even if it's a hypothetical 3 column wide glyph that fails to insert into a 2 column wide buffer. This PR would work in this situation, because without margins, on line wrap, the cursor will end up on column 0 and my assumption is that if a glyph fails to insert at column 0, it'll also fail to insert at any other column, so it needs to be dropped. I'm not sure how this would work with margins, because I don't quite understand how DECLRMM works yet (especially since I would've assumed that the cursor remains inside the margins during line wrap).

Reply:
Although, now that I think about it, if we're ever going to get to a point where we're writing out characters that occupy _more_ than two columns, this could be a problem. I think that might be more complicated to handle correctly though. I'm curious now how other terminals deal with that.

Reply:
To make a somewhat contrived, but still quite valid example, a category of characters that is combining but still considered to occupy a column are the characters from the "Spacing Mark" (Mc) category. One example is 온〮 (100 in middle Korean; U+C628 U+302E) which is a single grapheme and thus should optimally never be broken apart by us, but according to the asian width spec this grapheme is supposed to occupy 4 columns[^1].

[^1]: it's somewhat puzzling why they assigned 2 columns to U+302E - probably a typo on their end. This makes it a perfect example however.

Reply:
The complication is that the horizontal margins only apply to wrapping when you're also within the boundaries of the vertical margins. For example, consider these parameters:

* screen width is 80 columns
* horizontal margins are 79 to 80 (inclusive)
* vertical margins are 10 to 20
* cursor position is column 79, row 5

If you try to write out your 4-column character at that point, it's not going to fit. However, the cursor should wrap to column 1 on row 6, where a 4-column character _will_ fit, despite the fact that your starting point happened to be aligned with the left margin.

There may be other edge cases I haven't thought of, but that's the first one that springs to mind.